src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -303,6 +304,23 @@ export const handleDataSourceFetchError = (
   });
 };
 
+export const handleDataSourceViewError = (
+  changeState: (state: { showError: boolean }) => void,
+  notifications: ToastsStart,
+  failedDataSourceId: string,
+  defaultDataSource: string | null,
+  callback: () => Promise<void>
```

Comment:
Since we are defining a lot of props here, could we extract it into its own props class?

Reply:
Good idea, I will update, thanks for the suggestion